lib/App/Critique/Command/prune.pm

```diff
@@ -48,34 +48,10 @@ sub execute {
     my $filter;
 
     if ( my $f = $opt->filter ) {
-        $filter =  sub {
-            my $path     = $_[0]->path->stringify;
```

Comment:
Both these filters (`regex` and `no_violations`) end up taking a `App::Critique::Session::Path` object, so here we have some commonality we can pull up into a more generalized solution.